extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClientRecorder.java

```diff
@@ -58,7 +43,7 @@ public ReactiveMongoClient get() {
     public RuntimeValue<ReactiveMongoClient> getReactiveClient(String name) {
         return new RuntimeValue<>(
                 Arc.container()
-                        .instance(ReactiveMongoClient.class, literal(name + MongoClientBeanUtil.REACTIVE_CLIENT_NAME_SUFFIX))
+                        .instance(ReactiveMongoClient.class, literal(name))
```

Comment:
@loicmathieu this isn't tested at all?

Reply:
It is tested in both mongodb-client extension and mongodb-panache.
It surprised me a lot that this causes NPE.

I know that @geoand changes some stuff on the way MongoClient and ReactiveMongoClient are created.
On the issue, a reference to Camel has been done, maybe it's linked to the way Camel interract with the mongodb-client extension.

I'll request a reproducer so we can understand what's going on.

Reply:
+1 for Loic's comment

Reply:
We have a reproducer here: https://github.com/apache/camel-quarkus/issues/1608

Reply:
Basically the clients are referenced in this [camel mongo extension](https://github.com/apache/camel-quarkus/blob/master/extensions-support/mongodb/deployment/src/main/java/org/apache/camel/quarkus/support/mongodb/deployment/SupportMongoDBProcessor.java#L38). We are not yet using reactive clients. The failure happens when we create an application adding a named blocking client. I debugged and I saw that when the recorder calls the `getReactiveClient()` there is a null due to the fact that it does not find the bean (apparently because of that suffix): removing the suffix it works:

Log trace with error:
```
getReactiveClient(mongoClient1)
literal(mongoClient1reactive)
Handler (InstanceHandleImpl [bean=null, instance=null, creationalContext=null, parentCreationalContext=null, destroyed=false])
```
Some log trace when fixed:
```
getReactiveClient(mongoClient1)
literal(mongoClient1)
Handler (InstanceHandleImpl [bean=io.quarkus.mongodb.reactive.ReactiveMongoClient_004114c281073142f0021f5a81e92d036853efa8_Synthetic_Bean@69a375af, instance=io.quarkus.mongodb.impl.ReactiveMongoClientImpl@5ff1cf9, creationalContext=io.quarkus.arc.impl.CreationalContextImpl@49cf6c7c, parentCreationalContext=null, destroyed=false])
```

Reply:
> We have a reproducer mentioned here: [apache/camel-quarkus#1608](https://github.com/apache/camel-quarkus/issues/1608)

That was solved by another commit, now the issue is a bit different